common/key/key.go

```diff
@@ -36,34 +36,34 @@ type SubAccountCurrencyAsset struct {
 }
 
 // Pair combines the base and quote into a pair
-func (k *PairAsset) Pair() currency.Pair {
+func (k PairAsset) Pair() currency.Pair {
 	if k.Base == nil && k.Quote == nil {
 		return currency.EMPTYPAIR
 	}
 	return currency.NewPair(k.Base.Currency(), k.Quote.Currency())
 }
 
 // Pair combines the base and quote into a pair
-func (k *ExchangePairAsset) Pair() currency.Pair {
+func (k ExchangePairAsset) Pair() currency.Pair {
 	if k.Base == nil && k.Quote == nil {
 		return currency.EMPTYPAIR
 	}
 	return currency.NewPair(k.Base.Currency(), k.Quote.Currency())
 }
 
 // MatchesExchangeAsset checks if the key matches the exchange and asset
-func (k *ExchangePairAsset) MatchesExchangeAsset(exch string, item asset.Item) bool {
+func (k ExchangePairAsset) MatchesExchangeAsset(exch string, item asset.Item) bool {
 	return strings.EqualFold(k.Exchange, exch) && k.Asset == item
 }
 
 // MatchesPairAsset checks if the key matches the pair and asset
-func (k *ExchangePairAsset) MatchesPairAsset(pair currency.Pair, item asset.Item) bool {
+func (k ExchangePairAsset) MatchesPairAsset(pair currency.Pair, item asset.Item) bool {
```

Comment:
What's the reason for this? This has a tiny performance hit.

Reply:
It was for consistency of a readonly func to have all receivers be the same. Where is your benchmark with your allegations? 👮  I will change it back for you to save the  0.09ns
```
BenchmarkPair1
BenchmarkPair1-10    	26419788	        45.38 ns/op	       0 B/op	       0 allocs/op
BenchmarkPair2
BenchmarkPair2-10    	26526078	        45.29 ns/op	       0 B/op	       0 allocs/op
```

Reply:
It was 1.3... (something)ns to 1.879ns it was a tiny performance hit 😆. Consistency is fine you don't need to change 🤷 

<img width="365" alt="image" src="https://github.com/thrasher-corp/gocryptotrader/assets/9391715/4d6bacd0-340a-4969-8fb8-eb2f684e8191">

Picks for proof when test some random func

Reply:
Yeah I did a poor benchmark. The savings is 0.3ns for me and I've made all the functions a pointer receiver 🎉